supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesRuntimeAttributes.scala

```diff
@@ -52,18 +53,51 @@ object TesRuntimeAttributes {
       preemptibleValidation(backendRuntimeConfig),
     )
 
-  def apply(validatedRuntimeAttributes: ValidatedRuntimeAttributes, backendRuntimeConfig: Option[Config]): TesRuntimeAttributes = {
-    val docker: String = RuntimeAttributesValidation.extract(dockerValidation, validatedRuntimeAttributes)
-    val dockerWorkingDir: Option[String] = RuntimeAttributesValidation.extractOption(dockerWorkingDirValidation.key, validatedRuntimeAttributes)
-    val cpu: Option[Int Refined Positive] = RuntimeAttributesValidation.extractOption(cpuValidation(backendRuntimeConfig).key, validatedRuntimeAttributes)
-    val memory: Option[MemorySize] = RuntimeAttributesValidation.extractOption(memoryValidation(backendRuntimeConfig).key, validatedRuntimeAttributes)
-    val disk: Option[MemorySize] = RuntimeAttributesValidation.extractOption(diskSizeValidation(backendRuntimeConfig).key, validatedRuntimeAttributes)
-    val failOnStderr: Boolean =
-      RuntimeAttributesValidation.extract(failOnStderrValidation(backendRuntimeConfig), validatedRuntimeAttributes)
-    val continueOnReturnCode: ContinueOnReturnCode =
-      RuntimeAttributesValidation.extract(continueOnReturnCodeValidation(backendRuntimeConfig), validatedRuntimeAttributes)
-    val preemptible: Boolean =
-      RuntimeAttributesValidation.extract(preemptibleValidation(backendRuntimeConfig), validatedRuntimeAttributes)
+  def makeBackendParameters(runtimeAttributes: Map[String, WomValue],
+                            keysToExclude: Set[String],
+                            config: TesConfiguration): Map[String, String] = {
+
+    if (config.useBackendParameters)
+      runtimeAttributes
+        .filterKeys(k => !keysToExclude.contains(k))
+        .collect { case (key, strValue: WomString) => (key, strValue.value)}
+    else
+      Map.empty
+  }
+
+  def apply(validatedRuntimeAttributes: ValidatedRuntimeAttributes, rawRuntimeAttributes: Map[String, WomValue], config: TesConfiguration): TesRuntimeAttributes = {
```

Comment:
This `usedRuntimeAttribute` construction is ugly, but I couldn't find a better way. I think the way this code is laid out will help us remember to add any new attribute keys to the set.

Ideally we'd be able to create a collection of keys and extractors and do this iteratively, managing the set of used keys in a safer way, but due to the lack of compiler-checked relationship between keys and extracted types, I don't think that's possible (I've been down this road before). I would love to be proved wrong.

Reply:
I do find that more attractive and easier to read, but I worry that it makes it too easy to add a new validated attr and neglect to add it to the the `validations` set. I think a forceful comment will make me feel better, though.